source/02_algo/model_based_rl_eng.rst

```diff
@@ -0,0 +1,147 @@
+Model-Based Reinforcement Learning
+######################################
+
+
+Model-Based Reinforcement Learning (Model-Based RL) is an important branch of reinforcement learning. The agent learns an environmental dynamics model through information interacting with the environment, and then uses the model to generate data to optimize action strategies, or use the model for planning.
+Model-Based RL methods first let the agent learn the environment dynamics model from the data obtained by interacting with the environment, and then use the environment model to generate a large number of virtual samples, thereby reducing the number of interactions with the real environment. Therefore, the sample utilization efficiency can be naturally improved.
+
+.. image:: ./images/model-based_rl.png
+  :align: center
+  :scale: 55%
+
+The environment model can generally be abstracted mathematically into a state transition function and a reward function.
+After learning the environment model, ideally, the agent does not need to interact with the real environment, but only in the simulated environment, which maximizes the cumulative discount reward through the RL algorithm to obtain the optimal policy.
```

Comment:
**_In the ideal case, the agent does not need to interact with the real environment anymore after learning the dynamics model. The agent can now query the dynamics model to produce simulated samples, by which the cumulative discount reward can be maximized to obtain the optimal policy._**